evaluators/rank/NdcgEvaluator/__init__.py

```diff
@@ -14,9 +15,29 @@ def metric(self):
 
     def evaluate(self, actual: Sequence[Any], desired: Sequence[Any], *args, **kwargs) -> float:
         """"
-        :param actual: the matched document identifiers from the request as matched by jina indexers and rankers
-        :param desired: the expected documents matches ids sorted as they are expected
-        :return the evaluation metric value for the request document
+        :param actual: the scores predicted by the search system.
+        :param desired: the expected score given by user as groundtruth.
+        :return the evaluation metric value for the request document.
         """
+        actual_at_k = actual[:self.eval_at]
+        desired_at_k = actual[:self.eval_at]
+        if len(actual) < 2:
+            raise ValueError(f'Expecting gains with minimal length of 2, {len(actual)} received.')
+        dcg = self._compute_dcg(gains=actual_at_k)
+        idcg = self._compute_idcg(gains=desired_at_k)
+        if idcg == 0.0:
+            return 0.0
+        else:
+            return dcg/idcg
+
+    def _compute_dcg(self, gains):
+        """Compute discounted cumulative gain."""
         ret = 0.0
-        pass
+        for score, position in zip(gains[1:], range(2, len(gains) + 1)):
+            ret += score/log(position, 2)
+        return gains[0] + ret
+
+    def _compute_idcg(self, gains):
```

Comment:
I think these 2 can be made @static right?

Reply:
@JoanFM why was that?

Reply:
these 2 functions do not relly in any attribute of `self`?

Reply:
@JoanFM I get you point :) mostly we do this since it's not rely on any attribute of `self` and use it outside the class. Such as:
```suggestion
    @staticmethod
    def do_something(param):
            # do something, mostly utility purpose, without rely on self but the logic belongs to this class
            pass
          
```

And we can call it with `NDCGEvaluator.do_something(param = param)`

Reply:
since this function is called internally, i personally think it's useless to add the decorator.

Reply:
Or move it outside the class, as a helper function somewhere

Reply:
i think it make more sense to leave it inside the class, logically it should be inside.